ppx-lib/gen_js_api_ppx.ml

```diff
@@ -1580,5 +1605,9 @@ let mapper =
     Ast_mapper.structure = (fun _this str ->
       check_loc_mapper.Ast_mapper.structure
         check_loc_mapper
-        (mapper.Ast_mapper.structure mapper str))
+        (mapper.Ast_mapper.structure mapper str));
+    signature = (fun this s ->
+      (* run [str_of_sg] to mark `js.***` attributes as used. *)
+      let (_ : structure) = str_of_sg ~global_attrs:[] s in
```

Comment:
This call to `str_of_sg` breaks the `ppx` mode of `gen_js_api`: for example, every sum type needs to be annotated with a `js.*` attribute, etc. In fact, the `ppx` mode is not supposed to do anything on `.mli` files, see https://github.com/LexiFi/gen_js_api/blob/master/PPX.md for the details.

Perhaps this check can be moved to the `ppxlib` driver ?

Reply:
This is done